snuba/subscriptions/executor_consumer.py

```diff
@@ -404,6 +437,10 @@ def terminate(self) -> None:
 
     def join(self, timeout: Optional[float] = None) -> None:
         start = time.time()
+
+        # Commit all pending offsets
+        self.__throttled_commit(force=True)
```

Comment:
You might want to batch the individual commits happening here https://github.com/getsentry/snuba/pull/2543/files#diff-b81082d36647653087818cb560a918148759eb0e770ef1c04313589ea32167d7R454-R456 as well and then commit them altogether.

Reply:
I thought about this, but prefer to leave it as is. I think it's slightly simpler and it's harmless to call commit on each of the pending futures when we shut down. It's not possible for there to be a large number.

Reply:
Do you have a rough idea of how large the pending futures might be?

In general, when the consumers are shutting down, we should try to finish shut down as quickly as possible. Think about the subscriptions rebalance problems we started seeing recently. If the complete shutdown happens quickly, in theory we should have fewer rebalances.

Reply:
The theoretical maximum is the total concurrent queries (which we have defined as 24) divided by the number of executor replicas. So if we have 8 replicas, the max possible is 3.

Reply:
Ok. That's not a big number so we should be fine. I forgot that these were running queries and not processing ingest traffic.